### PR TITLE
feat: limit ability to delete chapter to the owner

### LINKED ISSUE #1217: Resolvers requiring Authorization decorators
To complete the role-based authorization the Authorized decorator needs to be applied to the following resolvers:

- [x] #1298 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/Events/resolver.ts#L480
- [x] #1332 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/Events/resolver.ts#L616 (chapter admin)
- [x] #1323 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/Events/resolver.ts#L651 (chapter admin)
- [x] #1346 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/Events/resolver.ts#L661
- [x] #1280 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/ChapterUser/resolver.ts#L45 (instance member)
- [x] #1280 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/ChapterUser/resolver.ts#L73 (instance member)
- [x] #1284 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/ChapterUser/resolver.ts#L183 (instance owner)
- [x] #1413 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/ChapterUser/resolver.ts#L212 (chapter admin)
- [x] #1416 https://github.com/freeCodeCamp/chapter/blob/65beab1cbb57e2d5d5eaca80fd3b8d997c6743c2/server/src/controllers/ChapterUser/resolver.ts#L209 (chapter admin)
- [x] #1416 https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/server/src/controllers/ChapterUser/resolver.ts#L257 (chapter admin)
- [ ] #1605 https://github.com/freeCodeCamp/chapter/blob/6669bec958bea3c18d3d18809250de2858289f39/server/src/controllers/Chapter/resolver.ts#L96 (instance owner)

The following is a reasonable implementation approach

1. Make sure there is a cypress test of the functionality that uses the lowest permission user (i.e. instance owner > chapter admin > instance member)
2. Add tests for the forbidden users e.g. https://github.com/freeCodeCamp/chapter/blob/bee4c91ce972bb50e3fb3185b33b5e6157a8903f/cypress/e2e/dashboard/sponsors.cy.js#L59-L68
3. Add the Authorized decorator
4. Fix what breaks (often the first test will fail because the query sends, say, `id` rather than `chapterId` or `eventId` and is rejected by `authChecker`)

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #1217

I have tried to login as admin and delete chapter, and got this 

image

there is no cypress test yet, I am planning to give it a try. 👍 

Edit: This pull request was made to test myself "if I can add permission successfully", if you have PR that finish the issue feel free to open it and I will close this.

in hindsight should have asked in the issue before opening the pull request

### CODE REVIEW

File: cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts [hunk 1]
```diff
@@ -73,4 +73,28 @@ describe('chapter edit dashboard', () => {
       cy.contains(chapterData.name);
     });
   });
+
+  it('deny admins to delete a chapter', () => {
```
Comment (ojeytonwilliams) [inline cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts:77]:
```suggestion
  it('only accepts chapter deletion requests from owners', () => {
```

File: cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts [hunk 2]
```diff
@@ -73,4 +73,28 @@ describe('chapter edit dashboard', () => {
...
+    const chapterId = 1;
+    cy.login(Cypress.env('JWT_ADMIN_USER'));
```
Comment (ojeytonwilliams) [inline cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts:79]:
```suggestion
    cy.login('admin@of.chapter.one);
```
The login function is a little more reasonable these days, you just pass it the email address you want to log in as.

File: cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts [hunk 3]
```diff
@@ -73,4 +73,28 @@ describe('chapter edit dashboard', () => {
...
+    cy.updateChapter(chapterId, chapterData).then((response) => {
+      expectToBeRejected(response);
+
+      cy.visit(`/dashboard/chapters/${chapterId}`);
+      cy.findByRole('button', { name: 'Delete Chapter' })
+        .click()
+        .findByRole('button', { name: 'Are you sure?' })
+        .click()
+        .findByRole('button', { name: 'Delete' })
+        .click();
```
Comment (ojeytonwilliams) [inline cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts:90]:
```suggestion
```
There's no need to lock down a broken UI with a test.  As well as the server rejecting the request, the user should not be able to make the request through the UI.

File: cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts [hunk 4]
```diff
@@ -73,4 +73,28 @@ describe('chapter edit dashboard', () => {
       cy.contains(chapterData.name);
     });
   });
+
+  it('deny admins to delete a chapter', () => {
+    const chapterId = 1;
+    cy.login(Cypress.env('JWT_ADMIN_USER'));
+
+    cy.updateChapter(chapterId, chapterData).then((response) => {
```
Comment (ojeytonwilliams) [inline cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts:81]:
```suggestion
    cy.deleteChapter(chapterId).then((response) => {
```

File: cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts [hunk 5]
```diff
@@ -73,4 +73,28 @@ describe('chapter edit dashboard', () => {
...
+        .findByRole('button', { name: 'Delete' })
+        .click();
+    });
+
+    cy.updateChapter(chapterId, chapterData).then((response) => {
+      expect(response.status).to.eq(200);
+
+      cy.visit(`/dashboard/chapters/${chapterId}`);
+      cy.contains(chapterData.name);
+    });
```
Comment (ojeytonwilliams) [inline cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts:98]:
```suggestion
    cy.login();
    cy.deleteChapter(chapterId).then((response) => {
      expect(response.status).to.eq(200);
      expect(response.body.errors).not.to.exist;
    });
```
We're only testing deletion... and I needed somewhere to put the owner request!

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts [hunk 6]
```diff
@@ -73,4 +73,19 @@ describe('chapter edit dashboard', () => {
       cy.contains(chapterData.name);
     });
   });
+
+  it('only accepts chapter deletion requests from owners', () => {
+    const chapterId = 1;
```
Comment (Sboonny) [inline cypress/e2e/dashboard/chapters/[chapterId]/edit.cy.ts:78]: Delete duplicate variable 

```suggestion
```

**Review by ojeytonwilliams — APPROVED** [review]
Nice, it all seems to be working!

LGTM :+1:

**Pull request merged**